### PR TITLE
feat: AWS CDK infrastructure (TDE-219)

### PR DESCRIPTION
**Pull request by blacha**
### Change Description:
Add the AWS `cdk` infrastructure for `topo-processor`

### Notes for Testing:
`nxp cdk deploy`

### CODE REVIEW

**Comment by paulfouquet** [comment]
Close it to open a new one so @blacha is not the owner of it (for review...)

File: infra/src/batch.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    );
+    instanceRole.addManagedPolicy(
+      ManagedPolicy.fromAwsManagedPolicyName("AmazonSSMManagedInstanceCore")
+    );
+
+    instanceRole.addToPrincipalPolicy(
+      new PolicyStatement({ resources: ["*"], actions: ["sts:AssumeRole"] })
+    );
+
+    Bucket.fromBucketName(
```
Comment (blacha) [inline infra/src/batch.ts:62]: Not sure we need this to have permission to a `megantestbucket`?

File: infra/src/batch.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+export class AwsBatchStack extends Stack {
+  public constructor(scope: Construct, id: string, props: BatchStackProps) {
+    super(scope, id, props);
+
+    const container = new DockerImageAsset(this, "BatchContainer", {
+      directory: props.container,
+    });
+    const image = ContainerImage.fromDockerImageAsset(container);
+
+    const vpc = Vpc.fromLookup(this, "AlbVpc", {
```
Comment (blacha) [inline infra/src/batch.ts:40]: nit: this vpc has nothing to do with ALBs

File: infra/src/batch.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    const computeEnvironment = new ComputeEnvironment(this, "BatchCompute", {
+      managed: true,
+      computeResources: {
+        instanceRole: instanceRole.roleName,
+        vpc,
+        type: ComputeResourceType.SPOT,
+        maxvCpus: 100,
+        minvCpus: 0,
+        desiredvCpus: 1,
+        instanceTypes: [InstanceType.of(InstanceClass.C5, InstanceSize.LARGE)],
```
Comment (blacha) [inline infra/src/batch.ts:82]: We should pass in a list of instance types upto say a 4xl? or possibly even bigger?

File: infra/src/submit.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
+import * as sdk from "aws-sdk";
+import * as ulid from "ulid";
+
+const batch = new sdk.Batch();
+
+const JobDefinitionArn = "";
```
Comment (blacha) [inline infra/src/submit.ts:6]: these two could be looked up by the `CfnOuput`

**Review by MDavidson17 — APPROVED** [review]

**Review by amfage — APPROVED** [review]

**Pull request merged**